BLUI-3594: Add notes about conducting design audits (#598)
brightlayer-ui/doc-it: src/docs/design/design-audit.mdx

import TopSeverity from '../../app/assets/design/design-audit/top-severity.png';
import HighSeverity from '../../app/assets/design/design-audit/high-severity.png';
import MediumSeverity from '../../app/assets/design/design-audit/medium-severity.png';
import LowSeverity from '../../app/assets/design/design-audit/low-severity.png';
import Feature from '../../app/assets/design/design-audit/feature.png';
import Slides from '../../app/assets/design/design-audit/ppt.svg';
import DoDont from '../../app/assets/design/design-audit/do-dont.svg';
import Educational1 from '../../app/assets/design/design-audit/educational-1.png';
import Educational2 from '../../app/assets/design/design-audit/educational-2.png';

<!-- audit critique review -->

# UI Design Audits

A design audit is a great way to ensure that your user interface design meets Brightlayer UI’s design standard and lines up with other Brightlayer UI-based applications. This process examines not only branding elements, but also the usability of products.

<ImageGrid
    images={[Banner]}
    regularWidth
    caption={'A typical UI audit will highlight problems and provide design recommendations.'}
/>

> 

Reviewer: ```suggestion
Do not wait until a product is released to look for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and the development effort to adjust the user interface afterwards can potentially take months.
```

> Do not wait until product releases look for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and development effort the user interface can take months to address the critical / major issues identified.

Reviewer: ```suggestion
Do not wait until a product is released to look for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and the development effort to adjust the user interface afterwards can potentially take months.
```

Reviewer 2: For "Do not wait until a product is released to look for design audits"

- Just throwing ideas here but it could state
"Request design audits in the early stage of product readiness or tangible minimal viable product has been reached."
- or
"Request design audits when the product is a tangible minimal viable product"
- or
"Do not request a design audit prior to a product release"

> Do not wait until a product is about to be released to request for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and development effort the user interface can take months to address the critical / major issues identified.

Reviewer: ```suggestion
Do not wait until a product is about to be released to request for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and it can take months to implement the critical / major issues identified.
```

I'm reading through this now in depth; I've added a suggestion above just because there's a grammar issue in the original.   Anything else that I notice I'll put a suggestion in for.

<TOC
    anchors={[
        { title: 'Introduction', hash: 'ui-design-audits' },
        { title: 'Prepare for an Audit' },
        { title: 'Request for a Design Audit' },
        { title: 'Issue Severity' },
        { title: 'How to Conduct Design Audits (For Auditors)' },
        { title: 'Understand Audit Results' },
        { title: 'Design Audit Examples' },
    ]}
/>

# Prepare for an Audit

## Update Packages

Brightlayer UI sometimes makes style updates to components and themes through our NPM packages. Cosmetic issues such as Material component styling can be easily addressed by updating software packages to the latest versions. Therefore, when possible, teams are encouraged to update their Brightlayer UI packages and other major packages to the latest version.

Please refer to our [developer resources page](/resources/developer) to learn about our latest package versions.

Custom component styling is discouraged, because the Brightlayer UI’s theme may be suppressed. If the team identifies any styling that should be included as part of the Brightlayer UI component theme, please contact us.

## Read Through Our Website

> Please read through our [style guides](/style) and [design pattern guidelines](/patterns). Many examples cited by the webpage came from design audits we did in the past and are representative of common mistakes we observed across multiple projects. Team members may be able to identify low-hanging fruits during this self-audit and address these problems ahead of the audit.

Reviewer: Many examples cited by "the webpage"

Which one?  Style guides or design pattern guidelines?

## Grant Auditors Direct Access

Whenever possible, teams are encouraged to create a safe environment for the auditor to test the project directly. This is to help the auditor explore user interactions and different workflow branches.

## Communicate with User Journey

User experience design is centered around users. Instead of talking about the technical nature of your infrastructure, teams should explain to auditors in terms of user journeys. Who will be using the application? What do they care about the most? How will the application solve this user’s problem?

## Communicate the Limit

If the product has been around for a long time, the team should mention to its auditor any legacy issues. Auditors should also be informed of other limitations, such as limited ethernet connection speed, mobile app-exclusive features, etc.

## Define a Proper Scope

One reason we recommend doing design audits periodically is so that product teams do not work alone for too long before they receive any external feedback. This also helps keep the audit scope focused on a few workflows or screens at a time.

> If the product is complex or has never been audited, consider arranging only a few key workflows or screens to be audited. A large scope will “dilute” auditors’ attention and tend to contain a lot of repetitive comments. The team shall select the part of the product most important to its users, or that can help teams learn the most from it.

Reviewer: ```suggestion
If the product is complex or has never been audited, consider arranging only a few key workflows or screens to be audited. A large scope will “dilute” auditors’ attention and tend to contain a lot of repetitive comments. The product team should select the part of the product that is either most important to its users, or that can help guide future suggestions for other pages.
```
The second part of the original sentence was difficult for me to understand so I rephrased it arbitrarily.

# Issue Severity

> Design audits identify issues of different severities. Severity ratings are determined by their impact on the users. Brightlayer UI recommend using the following severity rating.

Reviewer: ```suggestion
Design audits identify issues of different severities. Severity ratings are determined by their impact on the users. Brightlayer UI recommends using the following severity rating.
```

1. Auditors understand better why the product team designed a feature in a certain way.
2. Auditors get a good design exercise.
3. A new design placed next to its original design serves as a good side-by-side comparison.

## Summarize Takeaways

Auditors are encouraged to provide an overall summary of their audit findings. This provides teams quick takeaways that helps them estimate their effort in improving the UI design.

## Presentation

Auditors may tune their presentation depending on who they will be presenting to. For example, if the audiences possess a more technical background, consider explaining the styling in CSS notations; if they are more on the product design side, consider showing your raw design process and inviting them for a design critique.

Some team members might be daunted by the auditor’s authority in design; however, they have been with the project way longer than the auditor and likely know some legacy or technical reasons for why things are done in a certain way. Auditors should encourage teams to challenge the design audit result to foster a better discussion.

> Auditors need to remember to pause from time to time in their presentations for the audiences to catch up — likely not many people in the audience are used to thinking the way a UI designer thinks. Auditors should also ask one of the audience to take notes.

Reviewer: ```suggestion
Auditors need to remember to pause from time to time in their presentations for the audiences to catch up — likely not many people in the audience are used to thinking the way a UI designer thinks. Auditors should also ask one member of the audience to take notes.
```

> Recommend Next Steps

Reviewer: ```suggestion
## Recommend Next Steps
```

Before the end of their presentations, auditors should recommend the next steps for teams to act upon. If the auditor has a technical background, they may list a few items as easy-to-fix quick-win items for the developer. If there are major feature suggestions, the auditor may picture how they will fit into the product’s big picture, and what the R&D timeline would look like.

## Improve Brightlayer UI

> During the audit, if auditors noticed anything worth componentizing, or that the existing Brightlayer UI standard is too strict, please suggest it to us. Many design guidelines in Brightlayer UI today came from real project design audits we did.

Reviewer: ```suggestion
During the audit, if auditors noticed anything worth componentizing or that the existing Brightlayer UI standard is too strict, please suggest it to us. Many design guidelines in Brightlayer UI today came from real project design audits we did.
```

Reviewer: Some of my comments around removing the pauses comes down to _how_ we wants the docs to read, I guess more of a voice suggestion.  If this is a theme that we have going on other pages, where we add pauses between parts of our sentence, feel free to ignore those suggestions.

import TopSeverity from '../../app/assets/design/design-audit/top-severity.png';
import HighSeverity from '../../app/assets/design/design-audit/high-severity.png';
import MediumSeverity from '../../app/assets/design/design-audit/medium-severity.png';
import LowSeverity from '../../app/assets/design/design-audit/low-severity.png';
import Feature from '../../app/assets/design/design-audit/feature.png';
import Slides from '../../app/assets/design/design-audit/ppt.svg';
import DoDont from '../../app/assets/design/design-audit/do-dont.svg';
import Educational1 from '../../app/assets/design/design-audit/educational-1.png';
import Educational2 from '../../app/assets/design/design-audit/educational-2.png';

<!-- audit critique review -->

# UI Design Audits

A design audit is a great way to ensure that your user interface design meets Brightlayer UI’s design standard and lines up with other Brightlayer UI-based applications. This process examines not only branding elements, but also the usability of products.

<ImageGrid
    images={[Banner]}
    regularWidth
    caption={'A typical UI audit will highlight problems and provide design recommendations.'}
/>

> Do not wait until a product is about to be released to request for design audits. Design audits are not just about the cosmetic alignments, but focused more on the usability side, especially when the product team does not have a professional UI/UX designer. Design audits themselves can take weeks, depending on the scope, and it can take months to implement the top- / high-severity issues identified.

Reviewer 3: ```suggestion
Don't wait until a product is ready for release to request a design audit. Design audits are not just about the cosmetic alignments, but also focus on the overall usability side — this is especially important when your team does not have a professional UI/UX designer on staff. The design audit itself may take several weeks, depending on the scope, and implementing the recommended (high-severity) changes could take months.
```

Author: Don't wait until a product is ready for release to request a design audit. Design audits are not just about the cosmetic alignments, but also focus on the overall usability side — this is especially important when your team does not have a professional UI/UX designer on staff. The design audit itself may take several weeks, depending on the scope, and implementing the recommended changes could take months.

Author: ```suggestion
Don't wait until a product is ready for release to request a design audit. Design audits are not just about the cosmetic alignments, but also focus on the overall usability side — this is especially important when your team does not have a professional UI/UX designer on staff. The design audit itself may take several weeks, depending on the scope, and implementing the recommended changes could take months.
```

> This article is written for both audit requesters (“teams”) and auditors to serve as guidance for how design audits should be conducted around the Brightlayer UI design system.

Reviewer 3: ```suggestion
This page outlines the process for audit requesters (“teams”) and auditors and should serve as a guideline for how the Brightlayer UI audit process works.
```

Author: ```suggestion
This page outlines the process for audit requesters (“teams”) and auditors and should serve as a guideline for how UI design audits work.
```

<TOC
    anchors={[
        { title: 'Introduction', hash: 'ui-design-audits' },
        { title: 'Prepare for an Audit' },
        { title: 'Request for a Design Audit' },
        { title: 'Issue Severity' },
        { title: 'How to Conduct Design Audits (For Auditors)' },
        { title: 'Understand Audit Results' },
        { title: 'Design Audit Examples' },
    ]}
/>

# Prepare for an Audit

## Update Packages

> Brightlayer UI sometimes makes style updates to components and themes through our NPM packages. Cosmetic issues such as Material component styling can be easily addressed by updating software packages to the latest versions. Therefore, when possible, teams are encouraged to update their Brightlayer UI packages and other major packages to the latest version.

Reviewer 3: ```suggestion
Brightlayer UI sometimes makes style updates to components and themes through our NPM packages. Cosmetic issues such as Material component styling can be easily addressed by updating software packages to the latest versions. Therefore, when possible, teams are encouraged to update their Brightlayer UI packages and other major packages to the latest version prior to requesting an audit.
```

Please refer to our [developer resources page](/resources/developer) to learn about our latest package versions.

> Custom component styling is discouraged because the Brightlayer UI’s theme may be suppressed. If the team identifies any styling that should be included as part of the Brightlayer UI component theme, please contact us.

Reviewer 3: ```suggestion
We try to discourage custom component styling that overrides the Brightlayer UI theme — these can can cause problems when updating to newer versions. If the team identifies any styling that should be included as part of the Brightlayer UI component theme, please contact us or log a bug.
```

Author: ```suggestion
We discourage custom component styling that overrides the Brightlayer UI theme — these can cause problems when updating to newer versions. If the team identifies any styling that should be included as part of the Brightlayer UI component theme, please [contact us](/community/contactus) or log an issue.
```

## Read Through Our Website

> Please read through our [style guides](/style) and [design pattern guidelines](/patterns). Many examples cited by these guidelines came from design audits we did in the past and are representative of common mistakes we observed across multiple projects. Team members may be able to identify low-hanging fruits during this self-audit and address these problems ahead of the audit.

Reviewer 3: ```suggestion
Please read through our [style guides](/style) and [design pattern guidelines](/patterns). Many examples cited by these guidelines came from design audits we did in the past and are representative of common mistakes we observed across multiple projects. Team members may be able to identify low-hanging fruit during this self-audit and address these problems ahead of the full audit.
```

## Address Comments from Previous Audits

If the team has received audit comments in the past, team members should address these audit comments first before requesting a new one. In case audit comments cannot be addressed due to time / technical constraints, the team should inform their auditors about this.

# Request for a Design Audit

Ideally, a design audit is conducted periodically by someone outside the product team. Team members may [request](/community/contactus) a design audit from the Brightlayer UI team.

## Give a Demo

Team members are expected to walk their auditors through the application to help them understand the big picture. This includes all major workflows of the application.

## Grant Auditors Direct Access

Whenever possible, teams are encouraged to create a safe environment for the auditor to test the project directly. This is to help the auditor explore user interactions and different workflow branches.

## Communicate with User Journey

> User experience design is centered around users. Instead of talking about the technical nature of your infrastructure, teams should explain to auditors in terms of user journeys. Who will be using the application? What do they care about the most? How will the application solve this user’s problem?

Reviewer 3: ```suggestion
User experience design is centered around users. Instead of talking about the technical nature of your infrastructure, teams should explain their product in terms of user journeys. Who will use the application? What do they care about the most? How will the application solve this user’s problem?
```

> ## Communicate the Limit

Reviewer 3: ```suggestion
## Communicate the Limitations
```

> If the product has been around for a long time, the team should mention to its auditor any legacy issues. Auditors should also be informed of other limitations, such as limited ethernet connection speed, mobile app-exclusive features, etc.

Reviewer 3: ```suggestion
The team should mention to its auditor any legacy issues (e.g., some things may need to stay a certain way to meet user expectations in a legacy product) or other limitations, such as limited ethernet connection speed, mobile app-exclusive features, etc.
```

> ## Define a Proper Scope

Reviewer 3: ```suggestion
## Define the Scope
```

One reason we recommend doing design audits periodically is so that product teams do not work alone for too long before they receive any external feedback. This also helps keep the audit scope focused on a few workflows or screens at a time.

> If the product is complex or has never been audited, consider arranging only a few key workflows or screens to be audited. A large scope will “dilute” auditors’ attention and tend to contain a lot of repetitive comments. The product team should select the part of the product that is either most important to its users, or that can help guide future redesign efforts for other pages.

Reviewer 3: ```suggestion
If the product is complex or has never been audited, consider arranging only a few key workflows or screens to be audited. A large scope will “dilute” auditors’ attention and lead to many repetitive comments. The product team should select the part of the product that is either most important to its users or a representative area that can help guide future redesign efforts for other pages.
```

# Issue Severity

> Design audits identify issues of different severities. Severity ratings are determined by their impact on the users. Brightlayer UI recommends using the following severity rating.

Reviewer 3: ```suggestion
Design audits identify issues using multiple different severity levels. Severity ratings are determined by their impact on the users. Brightlayer UI recommends using the following severity ratings:
```

A **top severity** issue is a design flaw that prevents a user from completing a certain task.

<ImageGrid images={[TopSeverity]} regularWidth />

> A **high severity** issue greatly confuses or blocks its users, although eventually they might learn to tolerate it.

Reviewer 3: ```suggestion
A **high severity** issue greatly confuses or blocks the user, although eventually they might learn to tolerate it.
```

<ImageGrid images={[HighSeverity]} regularWidth />

A **medium severity** issue makes users hesitate, but they are usually still able to complete their task after some guesswork.

<ImageGrid images={[MediumSeverity]} regularWidth />

A **low severity** issue is cosmetic and does not interfere with functionality or usability.

<ImageGrid images={[LowSeverity]} regularWidth />

> In addition, auditors may suggest **features** as well. Features are good-to-have improvements; users often notice them in other Brightlayer UI apps.

Reviewer 3: ```suggestion
In addition, auditors may suggest **features** as well. Features are good-to-have improvements and may be based on designs seen in other Brightlayer UI apps.
```

<ImageGrid images={[Feature]} regularWidth />

> As hinted above, audit notes may also be tagged with whether an issue is about usability or branding. This further helps product teams think about usability systematically.

Reviewer 3: ```suggestion
Audit notes may also be tagged with an indicator of whether an issue is about usability or branding. This further helps product teams think about usability systematically.
```

Auditors are recommended to color code their design audit notes based on severity.

# How to Conduct Design Audits (For Auditors)

## Prepare Deliverables

> A full audit done by the Brightlayer UI team typically includes two parts: detailed notes that capture everything, and a presentation slide deck including only items worth discussing.

Reviewer 3: ```suggestion
A full audit done by the Brightlayer UI team typically includes two parts: detailed notes that capture everything, and a presentation slide deck including only items that merit further discussion.
```

Author: this further complicate the sentence

Reviewer 3: maybe 'worth discussing together'. I think I was trying to avoid the implication that all the other stuff is 'not worth discussing'

## Apply Brightlayer UI’s Standard

<ImageGrid
    images={[DoDont]}
    captionsUnderImages={[`You can find these color-coded design standards across Brightlayer UI's design guidelines.`]}
/>

There are two parts to Brightlayer UI’s design guidelines — [style guide](/style) and [design patterns](/patterns). Each of these describes a set of design standards, which are categorized as follows:

<RuleTable />

User interfaces must be judged under these criteria during design audits.

In addition, auditors may also employ [Heuristic Evaluation](https://www.nngroup.com/articles/how-to-conduct-a-heuristic-evaluation/) to check UI designs comprehensively.

## Educate the Team

To help teams develop a human-centered mindset, in addition to pointing out the issues, auditors must also seek to educate their product team to think like designers.

<ImageGrid
    images={[Educational1, Educational2]}
    captionsUnderImages={[
        `AVOID: Avoid giving the team an instruction without explaining the rationale behind it.`,
        `DO: Teach the team how this design decision is made.`,
    ]}
/>

> For larger products, instead of trying to cover everything in superficial detail, auditors may select 2-3 key interactions and examine them in depth. The auditors’ mission should not be to point out every single UI mistake in this case, as there are likely going to be follow-up UI audits in the future, but to teach the team how to avoid similar mistakes in the future.

Reviewer 3: ```suggestion
For larger products, instead of trying to cover everything in superficial detail, auditors may select 2-3 key interactions and examine them in depth. The auditors’ mission should not be to point out every single UI mistake, but to teach the team how to avoid similar mistakes in the future.
```

## Prototype Key Features

If time permits, auditors may select a few key features and prototype their design recommendation. This has a few advantages.

1. Auditors understand better why the product team designed a feature in a certain way.

> 2. Auditors get a good design exercise.

Reviewer 3: ```suggestion
2. Auditors get an excellent opportunity for design practice.
```

3. A new design placed next to its original design serves as a good side-by-side comparison.

## Summarize Takeaways

> Auditors are encouraged to provide an overall summary of their audit findings. This provides teams with quick takeaways that help them estimate their effort in improving the UI design.

Reviewer 3: ```suggestion
We also encourage auditors to provide an overall summary of their audit findings. This provides teams with quick takeaways that help them estimate their effort in improving the UI design.
```

## Presentation

> Auditors may tune their presentation depending on who they will be presenting to. For example, if the audiences possess a more technical background, consider explaining the styling in CSS notations; if they are more on the product design side, consider showing your raw design process and inviting them for a design critique.

Reviewer 3: ```suggestion
Auditors may fine-tune their presentation depending on the expected audience. For example, if the audiences is more technical, consider explaining the styling in CSS notations; if they are more on the product design side, consider showing your raw design process and inviting them for a design critique.
```

Author: ```suggestion
Auditors may fine-tune their presentation depending on the expected audience. For example, if the audience is more technical, consider explaining the styling in CSS notations; if they are more on the product design side, auditors may consider showing their raw design process and inviting them for a design critique.
```

> Some team members might be daunted by the auditor’s authority in design; however, they have been with the project way longer than the auditor and likely know some legacy or technical reasons for why things are done in a certain way. Auditors should encourage teams to challenge the design audit result to foster a better discussion.

Reviewer 3: ```suggestion
Some team members might feel intimidated by the design audit or feel that the auditor is always right (they aren't). Auditors should encourage teams to challenge the design audit result to foster a better discussion and learn more from the team who knows the product and its history more intimately.
```

> Auditors need to remember to pause from time to time in their presentations for the audiences to catch up — likely not many people in the audience are used to thinking the way a UI designer thinks. Auditors should also ask one member of the audience to take notes.

Reviewer 3: ```suggestion
Auditors need to remember to pause from time to time in their presentations for the audiences to catch up — it's likely that many people in the audience are not used to thinking the way a UI designer thinks. Auditors should also ask one member of the audience to take notes.
```

## Recommend Next Steps

Before the end of their presentations, auditors should recommend the next steps for teams to act upon. If the auditor has a technical background, they may list a few items as easy-to-fix quick-win items for the developer. If there are major feature suggestions, the auditor may picture how they will fit into the product’s big picture, and what the R&D timeline would look like.

## Improve Brightlayer UI

> During the audit, if auditors noticed anything worth componentizing or that the existing Brightlayer UI standard is too strict, please suggest it to us. Many design guidelines in Brightlayer UI today came from real project design audits we did.

Reviewer 3: ```suggestion
During the audit, if auditors noticed anything worth componentizing or an area where the existing Brightlayer UI standard is too strict, please get in touch with us for a discussion. Many design guidelines in Brightlayer UI today came from real project design audits we performed.
```

# Understand Audit Results

## Prioritize Issues

Teams must prioritize high-severity issues as they are blocking the application’s functions. These issues must be addressed before the next major release. Meanwhile, mid-severity issues can wait until the next major release, but should still be groomed into the backlog. Low-severity issues are typically considered technical debt.
High-frequency issues — issues that may not be severe but are exposed to users — should also be prioritized.

## Do Not Skip User Testing

> A design audit is not meant to replace user testing. While auditors can flag potential issues based on purely their experience in the design industry, they are not the end user. Teams should conduct user testing to drive major changes while using design audit as a supplemental tool.

Reviewer 3: ```suggestion
A design audit is not meant to replace user testing. While auditors can flag potential issues based on their experience in the design industry, they are not the end user. Teams should conduct user testing to drive major changes while using the design audit as a supplemental tool.
```

## Schedule the Next Audit

Typically, one design audit is enough to cover most issues in the current version of the user interface. Product teams may consider reaching out to another designer after the existing issues have been pointed out.

Whenever there are major changes to the user interface, product teams should conduct a new design audit to ensure they are still on track.

> # Design Audit Examples

Reviewer 3: I would leave this section out (or just say something like 'If you would like to see examples of audits from the past, reach out to us") until we have a better idea how we want to handle 'private' materials

Author: what's the concern? There are no business secrets in the URL

Reviewer 3: I'm not an IT expert, but I know they have had concerns in the past about exposing URL structures for where we host files, source code, etc. But the fact that this is already in a commit has exposed it anyway, so 🤷‍♂️  I guess it can stay until they tell us to remove it.

I would update the blurb to say "If you are an Eaton employee, you can view example audit files below (login required).

Author: > If outside-Eaton adopters contacted us asking for an example, would we have one to provide?

No... I can fake one but it will take quite some time to do
